core/src/main/scala/com/salesforce/op/ModelInsights.scala

```diff
@@ -799,8 +799,16 @@ case object ModelInsights {
       case m: RandomForestRegressionModel => Seq(m.featureImportances.toArray.toSeq)
       case m: GBTRegressionModel => Seq(m.featureImportances.toArray.toSeq)
       case m: GeneralizedLinearRegressionModel => Seq(m.coefficients.toArray.toSeq)
-      case m: XGBoostRegressionModel => Seq(m.nativeBooster.getFeatureScoreVector(featureVectorSize).toArray.toSeq)
-      case m: XGBoostClassificationModel => Seq(m.nativeBooster.getFeatureScoreVector(featureVectorSize).toArray.toSeq)
+      case m: XGBoostRegressionModel =>
+        Try(Seq(m.nativeBooster.getFeatureScoreVector(featureVectorSize).toArray.toSeq)) match {
+          case Success(contrib) => contrib
+          case _ => Seq.empty
+        }
+      case m: XGBoostClassificationModel =>
+        Try(Seq(m.nativeBooster.getFeatureScoreVector(featureVectorSize).toArray.toSeq)) match {
```

Comment:
Why does this need to be wrapped in try? Has this changed location or name? Please track down what is failing rather than allowing it to silently fail

Reply:
No, there's no change in location or name. However, there's something flaky about extracting model importance from a trained XGBoost model. In my downstream integration test, `nativeBooster.getFeatureScoreVector` occasionally failed due to `require(featureScore.nonEmpty, "Feature score map is empty")`. I can't reproduce this error in my experiments on real datasets (they don't fail at all and modelInsights was extracted just fine) so i'm not sure if it's a real issue...

Reply:
Yeah, I'm not sure we'd want to allow this. I think requiring feature importances is a good idea, and it would be better to understand why this is happening. Is something going wrong with the model training entirely, and this is the first symptom of a larger problem?

I'd start with adding some logging to the failing test and see if that helps pinpoint the issue.

Reply:
@Jauntbox @leahmcguire turns out if there's no useful feature, XGBoost would return an empty vector here, would that justify having a `Try` ?